docs/discover/index.rst

```diff
@@ -0,0 +1,213 @@
+.. _pyvo-discover:
+
+******************************************
+Global Dataset Discovery (`pyvo.discover`)
+******************************************
+
+One of the promises of the Virtual Observatory has always been that
+researchers can globally look for data sets (spectra, say).  In the
+early days of the VO, this looked relatively simple: Just enumerate all
+image (in those days: SIAP) services, send the same query to them, and
+then somehow shoehorn together their responses.
+
+In reality, there are all kinds of small traps ranging from hanging
+services to the sheer number of data collections. Hitting hundreds of
+internet sites takes quite a bit of time even when everything works.  In
+the meantime, the picture is further complicated by the emergence of
+additional protocols.  Images, for instance, can be published through
+SIA version 1, SIA version 2, and ObsTAP in 2024 – and in particular any
+combination of that.
+
+To keep global discovery viable, several techniques can be applied:
+
+* pre-select the services by using the service footprints in space,
+  time, and spectrum.
+* elide services serving the same data collections
+* filter duplicate responses before presenting them to the user.
+
+This is the topic of this sub-module.
+
+In early 2024, this is still in early development.
+
+
+Basic Usage
+===========
+
+The basic API for dataset discovery is through functions accepting
+constraints on
+
+* space (currently, a cone, i.e., RA, Dec and a radius in degrees),
+* spectrum (currently, a point in the spectrum as a spectral quantity),
+  and
+* time (an astropy.time.Time instance or a pair of them to denote an
+  interval).
+
+For instance::
+
+  from pyvo import discover
+  from astropy import units as u
+  from astropy import time
+
+  datasets, log = discover.images_globally(
+    space=(273.5, -12.1, 0.1),
+    spectrum=1*u.nm,
+    time=(time.Time('1995-01-01'), time.Time('1995-12-31')))
+  print(datasets)
+
+The function returns a pair of lists.  ``datasets`` is a list of
+`~pyvo.discover.image.ImageFound` instances.  This is the (potentially
```

Comment:
Is this supposed to be public API? If yes, is the implementation detail of the `image` important, or can we flatten things out and expose it in `discover`?

Reply:
People are supposed to look inside ImageFound-s, and hence I suppose it needs to be public API.  I have changed as much.